### PR TITLE
Feature/sw 5 02/61 children child class controller

### LINKED ISSUE #61: Children / Child class controller

### PR DESCRIPTION
**Pull request by JacobInberg**
# Description

Create a child and classroom endpoint as well as add a child to a given class, pincode for users endpoint with update, check if pincode is correct as well as check if the user has a pincode

Fixes #61 

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version:
* Dart version:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: FoodplannerApi/Controller/ChildrensController.cs [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
     
+    public ChildrensController(IChildrenService childrenService){
+        _childrenService = childrenService;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetAll(){
+        var children = await _childrenService.GetAllChildrenAsync();
+        return Ok(children);
+    }
```
Comment (JacobStokholms) [inline FoodplannerApi/Controller/ChildrensController.cs:20]: Skal nok ikke bruge alle børn, men kun hente en klasse ad gangen

File: FoodplannerModels/Account/UserCredsDTO.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@ namespace FoodplannerModels.Account;
 
 public class UserCredsDTO
 {
+    public required int Id { get; set; }
```
Comment (JacobStokholms) [inline FoodplannerModels/Account/UserCredsDTO.cs:7]: Brug JWT til at retreive ID istedet

File: FoodplannerApi/Controller/UsersController.cs [hunk 3]
```diff
@@ -56,10 +56,52 @@ public async Task<IActionResult> Login([FromBody] Login user){
...
+            }
+            return BadRequest();
+            //return BadRequest(new ErrorResponse {Message = ["Forkert pinkode"]});
+        } catch (InvalidOperationException e){
+            return BadRequest(new ErrorResponse {Message = [e.Message]});
+        }
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> HasPinCode(int id) {
```
Comment (JacobStokholms) [inline FoodplannerApi/Controller/UsersController.cs:98]: Brug JWT til at retreive ID istedet

File: FoodplannerServices/Account/UserService.cs [hunk 4]
```diff
@@ -53,23 +53,49 @@ public async Task<int> DeleteUserAsync(int id){
...
+
+    public async Task<string> UpdateUserPinCodeAsync(string pinCode, int id){
+        if (pinCode.ToString().Length != 4){
+            throw new InvalidOperationException("Pinkode skal være 4 cifre");
+        }
+
+        var pincode = await _userRepository.UpdatePinCodeAsync(pinCode, id);
+
+        return pincode;
+    }
```
Comment (JacobStokholms) [inline FoodplannerServices/Account/UserService.cs:83]: Kan vi bruge Bcrypt til at cryptere pin-koden

File: FoodplannerServices/Account/UserService.cs [hunk 5]
```diff
@@ -53,23 +53,49 @@ public async Task<int> DeleteUserAsync(int id){
...
+        if (pinCode.ToString().Length != 4){
+            throw new InvalidOperationException("Pinkode skal være 4 cifre");
+        }
+
+        var pincode = await _userRepository.UpdatePinCodeAsync(pinCode, id);
+
+        return pincode;
+    } 
+
+    public async Task<string> GetUserByIdAndPinCodeAsync(int id, string pinCode){
```
Comment (JacobStokholms) [inline FoodplannerServices/Account/UserService.cs:85]: BCrypt skal så også bruges her.

**Review by JacobStokholms — CHANGES_REQUESTED** [review]

File: FoodplannerApi/Controller/ChildrensController.cs (hunk 1 above)
Comment (Dressi123) [reply FoodplannerApi/Controller/ChildrensController.cs]: den skal bruges i frontend ved oprettelse af barn og her skal man bruge alle klasser.

File: FoodplannerServices/Account/UserService.cs (hunk 5 above)
Comment (JacobInberg) [reply FoodplannerServices/Account/UserService.cs:84]: Er tilføjet nu

File: FoodplannerModels/Account/UserCredsDTO.cs (hunk 2 above)
Comment (JacobInberg) [reply FoodplannerModels/Account/UserCredsDTO.cs]: Er fixet

File: FoodplannerApi/Controller/UsersController.cs (hunk 3 above)
Comment (JacobInberg) [reply FoodplannerApi/Controller/UsersController.cs]: Er fixet

File: FoodplannerServices/Account/UserService.cs (hunk 4 above)
Comment (JacobInberg) [reply FoodplannerServices/Account/UserService.cs]: Er tilføjet

File: FoodplannerApi/Controller/ChildrensController.cs (hunk 1 above)
Comment (JacobInberg) [reply FoodplannerApi/Controller/ChildrensController.cs]: Skulle men den er fjernet

**Review by JacobStokholms — APPROVED** [review]

**Pull request merged**